packages/video_player/video_player/lib/src/web_vtt.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'closed_caption_file.dart';
+import 'package:html/parser.dart';
+
+/// Represents a [ClosedCaptionFile], parsed from the WebVTT file format.
+/// See: https://en.wikipedia.org/wiki/WebVTT
+class WebVTTCaptionFile extends ClosedCaptionFile {
+  /// Parses a string into a [ClosedCaptionFile], assuming [fileContents] is in
+  /// the WebVTT file format.
+  /// * See: https://en.wikipedia.org/wiki/WebVTT
+  WebVTTCaptionFile(this.fileContents)
+      : _captions = _parseCaptionsFromWebVTTString(fileContents);
+
+  /// The entire body of the VTT file.
+  final String fileContents;
+
+  @override
+  List<Caption> get captions => _captions;
+
+  final List<Caption> _captions;
+}
+
+List<Caption> _parseCaptionsFromWebVTTString(String file) {
+  final List<Caption> captions = <Caption>[];
+  int number = 1;
+  for (List<String> captionLines in _readWebVTTFile(file)) {
+    if (captionLines.length < 2) continue;
+
+    // Caption has header
+    bool hasHeader = captionLines.length > 2;
+    if (hasHeader) {
+      number = int.parse(captionLines[0]);
+    }
+
+    final int captionNumber = number;
+    final _StartAndEnd startAndEnd = _StartAndEnd.fromWebVTTString(
+        hasHeader ? captionLines[1] : captionLines[0]);
```

Comment:
nit: Flutter style guide prefers adding a trailing comma here and a new line before the close paren

```dart
final _StartAndEnd startAndEnd = _StartAndEnd.fromWebVTTString(
  hasHeader ? captionLines[1] : captionLines[0],
);
```

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation